File: src/products/product-builder.ts
#!/usr/bin/env node
import "source-map-support/register";
import * as cdk from "@aws-cdk/core";
import { ProductTemplate } from "./product.model";

export const synthProduct = (product: ProductTemplate) => {
  const app = new cdk.App({
    runtimeInfo: false,
    stackTraces: false,
    treeMetadata: false,
  });

  new product.stack(app, product.name);

  app.synth();

  return product.name;
};

File: python/src/products/product_builder.py
#!/usr/bin/env python3
from aws_cdk import core as cdk
from products.product_model import ProductTemplate


class SynthProduct():
    productName: str

    def __init__(self, product: ProductTemplate) -> None:
        app = cdk.App(
            runtime_info=False,
            stack_traces=False,
            tree_metadata=False
        )
        product.stack(app, product.name)
        app.synth()
        self.productName = product.name
